### PR TITLE
ref(ui): Convert stacktracelink modal to FC

### PR DESCRIPTION
**Pull request by scttcper**

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/components/events/interfaces/frame/stacktraceLinkModal.tsx [hunk 1]
```diff
@@ -97,108 +94,102 @@ class StacktraceLinkModal extends Component<Props, State> {
...
+              inline={false}
+              flexibleControlStateSize
+              stacked
+              name="source-code-input"
+              type="text"
+              value={sourceCodeInput}
+              onChange={onHandleChange}
+              placeholder={t(
+                `https://github.com/helloworld/Hello-World/blob/master/${filename}`
               )}
```
Comment (priscilawebdev) [inline static/app/components/events/interfaces/frame/stacktraceLinkModal.tsx:126]:
```suggestion
              placeholder={
                `https://github.com/helloworld/Hello-World/blob/master/${filename}`
              }
```

File: static/app/components/events/interfaces/frame/stacktraceLinkModal.tsx [hunk 2]
```diff
@@ -97,108 +94,102 @@ class StacktraceLinkModal extends Component<Props, State> {
...
+          <SourceCodeInput>
+            <StyledInputField
+              inline={false}
+              flexibleControlStateSize
+              stacked
+              name="source-code-input"
+              type="text"
+              value={sourceCodeInput}
+              onChange={onHandleChange}
+              placeholder={t(
```
Comment (priscilawebdev) [inline static/app/components/events/interfaces/frame/stacktraceLinkModal.tsx:124]: we do not need to translate here

File: static/app/components/events/interfaces/frame/stacktraceLinkModal.tsx [hunk 3]
```diff
@@ -1,58 +1,55 @@
...
       provider,
-      organization: this.props.organization,
+      organization,
     });
   }
 
-  handleSubmit = async () => {
-    const {sourceCodeInput} = this.state;
-    const {api, closeModal, filename, onSubmit, organization, project} = this.props;
+  const handleSubmit = useCallback(async () => {
```
Comment (priscilawebdev) [inline static/app/components/events/interfaces/frame/stacktraceLinkModal.tsx:52]: are you sure we should add `async` here?

**Review by priscilawebdev — APPROVED** [review]
I left some comments but they are unrelated to the conversion

File: static/app/components/events/interfaces/frame/stacktraceLinkModal.tsx (hunk 1 above)
Comment (scttcper) [reply static/app/components/events/interfaces/frame/stacktraceLinkModal.tsx:126]: nice catch 👍

File: static/app/components/events/interfaces/frame/stacktraceLinkModal.tsx (hunk 3 above)
Comment (scttcper) [reply static/app/components/events/interfaces/frame/stacktraceLinkModal.tsx:52]: we probably don't need useCallback

**Pull request merged**